adminResetPassword.php

```diff
@@ -0,0 +1,118 @@
+
+    $loggedIn = false;
+    $accessLevel = 0;
+    $userID = null;
+    if (isset($_SESSION['_id'])) {
+        $loggedIn = true;
+        // 0 = not logged in, 1 = standard user, 2 = manager (Admin), 3 super admin (TBI)
+        $accessLevel = $_SESSION['access_level'];
+        $userID = $_SESSION['_id'];
+    }
+    // Require admin privileges
+    if ($accessLevel < 2) {
+        header('Location: login.php');
+        die();
+    }
+
+    // Get all users from dbUsers table except for vmsroot
+    // This list of users will be used to populate the dropdown menu in the HTML form
+    require_once('database/dbUsers.php');
+    require_once('domain/User.php');
+    $users = get_all_users();
+
+    // if users is equal to false (meaning no users were retrieved from the database), redirect to the dashboard
+    if (!$users) {
+        header('Location: index.php?noUsers');
+        die();
+    }
+
+    if ($_SERVER["REQUEST_METHOD"] == "POST") {
+        require_once('include/input-validation.php');
+        require_once('database/dbUsers.php');
+        require_once('domain/User.php');
+        $args = sanitize($_POST, null);
+        $required = array(
+            "user_dropdown", "new_password" // Required fields for the form
+        );
+        $error1 = false;
+
+        if (!wereRequiredFieldsSubmitted($args, $required)) {
+            $error1 = true; // Form submission contained unexpected input; alert user and stay on this page
+        } else {
+            // Create new user with the values from args
+            $id = $args['user_dropdown'];
+
+            // Check that new password is different from current password
+            $user = retrieve_user($id);
+            if (password_verify($args['new_password'], $user->get_password())) {
+                $error2 = true; // new and old passwords are the same
+            } else {
+                // Change the user's password and redirect admin to the dashboard
+                $newPassword = password_hash($args['new_password'], PASSWORD_BCRYPT);
+                $result = change_password($id, $newPassword);
+                if (!$result) {
+                    // If the password change fails, alert the admin and redirect to the dashboard
+                    header('Location: index.php?pcFail');
+                } else {
+                    // Password change was successful, alert the admin and redirect to the dashboard
+                    echo '<script>document.location = "index.php?pcSuccess";</script>';
+                }
+            }
+        }
+    }
+    $date = null;
+
+?>
+<!DOCTYPE html>
+<html>
+<head>
+    <?php require_once('universal.inc') ?>
+    <title>BBBS | Reset User Password</title>
+</head>
+<body>
+<?php require_once('header.php') ?>
+<h1>Reset User Password</h1>
+<main class="date">
+    <!-- Error messages -->
+    <?php if (isset($error1)): ?>
+        <p class="error-toast">Your form submission contained unexpected input.</p>
```

Comment:
great addition of various error messages to let the user know what issue they're running into exactly when it comes to insufficient input.